### PR TITLE
AzureStorage-Enrichment

### PR DESCRIPTION
**Pull request by rushriva**
Required items, please complete
   
   Change(s):
   - Added analytical rule and workbook to Azure storage solution

   Reason for Change(s):
   - Azure Storage Solution Enrichment

   Version Updated:
   - NA
   
   Testing Completed:
   - Tested in local subscription

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @rushriva One of the check is failing for the pr,
Can you please resolve the following

**Comment by v-atulyadav** [comment]
Looking into this validation failures.

File: Solutions/Azure Storage/Analytic Rules/Detection of data exfiltration activities from Azure storage.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    | where FailureEndTime < SuccessStartTime
+    | project User, EventStartTime = FailureStartTime, EventEndTime = SuccessEndTime, IpAddresses
+customDetails:
+  EventStartTime: FailureStartTime
+  EventEndTime: SuccessEndTime
+  IPAddresses: IpAddresses
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: FullName
```
Comment (shainw) [inline Solutions/Azure Storage/Analytic Rules/Detection of data exfiltration activities from Azure storage.yaml:48]: Don't use legacy entity mapping any longer, change to Name + NTDomain or another strong identifier pair - https://learn.microsoft.com/en-us/azure/sentinel/entities-reference#user-account

File: Solutions/Azure Storage/Analytic Rules/Detection of data exfiltration activities from Azure storage.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        columnName: User
+version: 1.0.1
```
Comment (shainw) [inline Solutions/Azure Storage/Analytic Rules/Detection of data exfiltration activities from Azure storage.yaml:50]: First version should be 1.0.0, right?

File: Solutions/Azure Storage/Analytic Rules/Detection of data exfiltration activities from Azure storage.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+name: Detection of data exfiltration activities from Azure storage
+description: |
+  'Detection of data exfiltration activities from Azure storage accounts, based on anomaly detection mechanism.'
+severity: Medium
+status: Available
+requiredDataConnectors:
+  - connectorId: SalesforceServiceCloud
+    dataTypes:
+      - SalesforceServiceCloud
+queryFrequency: 20m
```
Comment (shainw) [inline Solutions/Azure Storage/Analytic Rules/Detection of data exfiltration activities from Azure storage.yaml:11]: 20m, seems closer to an NRT analytic.  If that is the case, then we should change it to that type.  If not, then have it query over a longer period of time and simply bin(TimeGenerated, 20m) the results so your failureCountThreshold still applies.

https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/binfunction

**Review by shainw — CHANGES_REQUESTED** [review]

**Comment by v-prasadboke** [comment]
Hello @shainw any updates

**Comment by v-prasadboke** [comment]
Hello @vakohl any feedback on this

**Comment by v-prasadboke** [comment]
Hello @vakohl please provide your update on this

**Comment by v-prasadboke** [comment]
Hello @vakohl waiting for your feedback

**Comment by v-prasadboke** [comment]
Hello @vakohl waiting for your feddback

**Comment by v-prasadboke** [comment]
Hello @vakohl waiting for your feedback

**Comment by v-prasadboke** [comment]
Hello @vakohl waiting for your feedback

**Comment by v-prasadboke** [comment]
Hello @vakohl shain has made some changes

**Pull request closed without merging**